readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -209,9 +216,15 @@ def build(self):
             '.',
             self.sphinx_build_dir,
         ])
+        config_file = (
+            self.config.sphinx.configuration or
+            self.project.conf_file(self.version.slug)
+        )
```

Comment:
Same here. And I feel like we shouldn't be guessing here. If the config file was specified and it exists, that's when we should set the config file path. Likewise, if we're generating a config, we set a config file path when we find out the user doesn't have a config and didn't specify a config file path.

This will cut down on magical decisions that tend to hide errors from us.